rosbag2_compression/test/rosbag2_compression/test_zstd_compression.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include <fstream>
+#include <string>
+
+#include "rosbag2_storage/filesystem_helper.hpp"
+#include "rosbag2_test_common/temporary_directory_fixture.hpp"
+
+#include "rosbag2_compression/zstd_compression.hpp"
+
+using namespace testing;  // NOLINT
+using namespace rosbag2_storage;  // NOLINT
+using namespace rosbag2_test_common;  // NOLINT
+
+class FilesystemHelperFixture : public TemporaryDirectoryFixture {};
+
+void create_garbage_file(const std::string & uri)
+{
+  std::ofstream out(uri);
+  for (int i = 0; i < 984467; i++) {
+    out << "GarbageGarbageGarbageGarbage";
+  }
+  out << "test";
+  out.close();
+}
+
+TEST_F(FilesystemHelperFixture, zstd_compress_file_uri)
+{
+  rclcpp::init(0, nullptr);
+  // 27MiB file
+  auto uri = FilesystemHelper::concat({temporary_dir_path_, "file1.txt"});
```

Comment:
This needs a fixture to make sure we're not clobbering the disk every time this runs.